staging/mausezahn.c

```diff
@@ -710,11 +719,20 @@ int getopts (int argc, char *argv[])
 				(ipv6_mode && get_ip6_range_dst(tx.ip_dst_txt, l)) || // returns 1 when no range has been specified
 				(!ipv6_mode && get_ip_range_dst(tx.ip_dst_txt)))
 			{
-				// name2addr4 accepts a DOTTED DECIMAL ADDRESS or a FQDN:
-				if (ipv6_mode)
+				// name2addr{4, 6} accept a valid IP address or a FQDN:
+				if (ipv6_mode) {
 					tx.ip6_dst = libnet_name2addr6 (l, tx.ip_dst_txt, LIBNET_RESOLVE);
-				else
-					tx.ip_dst = libnet_name2addr4 (l, tx.ip_dst_txt, LIBNET_RESOLVE);
+					if (strncmp((char*)&tx.ip6_dst,(char*)&in6addr_error,sizeof(in6addr_error))==0) {
+					  fprintf(stderr, "Failed to set destination IPv6 address. Please check if destination is set to a valid IPv6 address.\n");
+					  return 1;
+					}
+				}
+				else {
+					if ((tx.ip_dst = libnet_name2addr4 (l, tx.ip_dst_txt, LIBNET_RESOLVE)) == -1){
+					  fprintf(stderr, "Failed to set destination IPv4 address. Please check if destination is set to a valid IPv4 address.\n");
+					  return 1;
+					}
+			  }
 			}
```

Comment:
This situation seems a bit tricky to me, and I wanted to get some more input before going further.

Here is what I know so far:

* `libnet` itself defines these checks for error conditions and uses them in [sample](https://github.com/sam-github/libnet/blob/master/libnet/sample/synflood6.c#L93-L97) [code](https://github.com/sam-github/libnet/blob/master/libnet/sample/synflood6.c#L93-L97)

* The limitation where the error condition corresponds to a global broadcast address seems to be [known to some extent](https://books.google.com/books?id=lgh-akRVTuMC&pg=PA43&dq=%22is+actually+the+IP+address+255.255.255.255%22&hl=en&sa=X&ved=0ahUKEwiSztnn4YDRAhVM_IMKHeCCBfYQ6AEIGzAA#v=onepage&q=%22is%20actually%20the%20IP%20address%20255.255.255.255%22&f=false)

* As a result, it is impossible to determine whether the return value of the `libnet_name2addr` functions is a global broadcast because the user wanted to set this, or because the library is returning an error. For example, if the user is using something like `-6 -A 4.4.4.4`, the value returned by `libnet` is still going to look like a global broadcast address. I verified that this does indeed happen – so my initial statement at the top of the PR, “`mausezahn` fails silently” wasn't quite correct. It should have been “tries sending to the global broadcast address when you get invalid input.”

* However, I can't call `inet_pton` straight on the _input_ (say `tx.ip_dst_txt`) because it might be a hostname. I was depending upon `libnet` to do this for me and give me something that was guaranteed to be an IP address.

In view of this, maybe it would be good to just explicitly exclude the case where the user was putting in global broadcast addresses from validation, but do it for all other addresses. So, for example, something like `-6 -A 255.255.255.255` wouldn't throw an error, but `-6 3.3.3.3` would.

Thoughts?